src/module/actor/data-model-character.js

```diff
@@ -0,0 +1,204 @@
+import OseDataModelCharacterAC from "./dataModelClasses/OseDataModelCharacterAC";
+import OseDataModelCharacterEncumbrance from "./dataModelClasses/OseDataModelCharacterEncumbrance";
+import OseDataModelCharacterMove from "./dataModelClasses/OseDataModelCharacterMove";
+import OseDataModelCharacterScores from "./dataModelClasses/OseDataModelCharacterScores";
+import OseDataModelCharacterSpells from "./dataModelClasses/OseDataModelCharacterSpells";
+
+const getItemsOfActorOfType = (actor, filterType, filterFn = null) =>
+  actor.items
+    .filter(({type}) => type === filterType)
+    .filter(filterFn ? filterFn : () => true);
+
+export default class OseDataModelCharacter extends foundry.abstract.DataModel {
+  prepareDerivedData() {
+    this.scores = new OseDataModelCharacterScores(this.scores);
+
+    this.encumbrance = new OseDataModelCharacterEncumbrance(
+      game.settings.get(game.system.id, "encumbranceOption"),
+      this.encumbrance.max,
+      [...this.parent.items],
+      game.settings.get(game.system.id, "significantTreasure"),
+    );
+
+    this.movement = new OseDataModelCharacterMove(
+      this.encumbrance,
+      this.config.movementAuto,
+      this.movement.base
+    );
+
+    this.ac = new OseDataModelCharacterAC(
+      false,
+      [...getItemsOfActorOfType(this.parent, 'armor', (a) => a.system.equipped)],
+      this.scores.dex.mod,
+      this.ac.mod,
+    );
+
+    this.aac = new OseDataModelCharacterAC(
+      true,
+      getItemsOfActorOfType(this.parent, 'armor', (a) => a.system.equipped),
+      this.scores.dex.mod,
+      this.aac.mod,
+    );
+
+    this.spells = new OseDataModelCharacterSpells(
+      this.spells,
+      this.#spellList
+    )
+  }
+
+  // @TODO set up schema
+  // @todo define schema
```

Comment:
Looks like your comment in the PR specified this is still WIP.

Reply:
The double `todo` message is a bit much :upside_down_face:  But yeah, the constraints part of the schema isn't done, mainly because I don't know the constraints yet (aside from stuff like, obviously most numbers shouldn't go below 0)